src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1457,6 +1461,68 @@ public void setConcurrentCompactors(int value)
         CompactionManager.instance.setConcurrentCompactors(value);
     }
 
+    public String getAuthenticator()
+    {
+        return DatabaseDescriptor.getAuthenticator().getClass().getName();
+    }
+
+    public void setAuthenticator(String value)
```

Comment:
you do not count on a fact that some implementations might have additional parameters (will be done for all stuff like that in CASSANDRA-19946). Once that one is in, you need to rework this in such a way that these methods will be able to take parameters into account as well.

Reply:
Until CASSANDRA-19946 lands I don't think it needs to be blocker for this